ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -128,148 +131,157 @@ public MaterializedWindowedTable windowed() {
     return Optional.of(intermediate);
   }
 
+  private void updateConsistencyVector(
+      final Optional<ConsistencyOffsetVector> consistencyVector,
+      final Optional<Position> position
+  ) {
+    if (position.isPresent() && consistencyVector.isPresent()) {
+      consistencyVector.get().updateFromPosition(position.get());
+    }
+  }
+
   final class KsqlMaterializedTable implements MaterializedTable {
 
-    private final MaterializedTable table;
+    private final StreamsMaterializedTable table;
 
-    KsqlMaterializedTable(final MaterializedTable table) {
+    KsqlMaterializedTable(final StreamsMaterializedTable table) {
       this.table = requireNonNull(table, "table'");
     }
 
     @Override
-    public KsMaterializedQueryResult<Row> get(final GenericKey key, final int partition) {
-      if (transforms.isEmpty()) {
-        return table.get(key, partition);
-      }
+    public KsqlMaterializedQueryResult<Row> get(
+        final GenericKey key,
+        final int partition,
+        final Optional<ConsistencyOffsetVector> consistencyVector
+    ) {
+      final Optional<Position> position = consistencyVector
+          .map(offsetVector -> Position.fromMap(offsetVector.getOffsetVector()));
+      final KsMaterializedQueryResult<Row> result = table.get(key, partition, position);
+      updateConsistencyVector(consistencyVector, result.getPosition());
```

Comment:
Update the consistency token with the position we get from streams after executing the query

Reply:
Is the "Optional<ConsistencyOffsetVector> consistencyVector" also used an output param (input anyway)? If so why, don't we pass the consistency vector as part of the iterator?

Reply:
I don't understand the question :(